libtesting-ui/src/main/java/com/mapbox/navigation/testing/ui/http/MockWebServerRule.kt

```diff
@@ -64,4 +73,19 @@ class MockWebServerRule : TestWatcher() {
         requestHandlers.clear()
         webServer.shutdown()
     }
+
+    private var previousPort: Int? = null
+
+    fun stop() {
+        previousPort = webServer.port
+        webServer.shutdown()
+    }
+
+    suspend fun restart() {
+        withContext(Dispatchers.IO) {
+            webServer = MockWebServer()
+            initDispatcher()
+            webServer.start(previousPort!!)
+        }
+    }
```

Comment:
@dzinad , @DzmitryFomchyn , I can't say I like current implementation. If you know a better way to emulate "not internet connection", please suggest